overlays/attribute-ordering.nix

```
@@ -30,7 +30,7 @@ let
   checkDerivation = drvArgs: drv:
     let
       getAttrPos = attr: (builtins.unsafeGetAttrPos attr drvArgs);
-      getAttrLine = attr: (getAttrPos attr).line;
+      getAttrLine = attr: (getAttrPos attr).line or 0;
 
       drvAttrs = builtins.sort (a: b: getAttrLine a < getAttrLine b) (builtins.attrNames drvArgs);
```

I think I would rather do something like

```suggestion
      drvAttrs = builtins.sort (a: b: getAttrLine a < getAttrLine b) (builtins.filter (attr: getAttrPos attr != null) (builtins.attrNames drvArgs));
```

otherwise, we might get warnings if e.g. `meta` does not have location.

---

Maybe even

```suggestion
      drvAttrs = lib.pipe drvArgs [
        builtins.attrNames
        # Certain functions like mapAttrs can produce attributes without associated position.
        (builtins.filter (attr: getAttrPos attr != null))
        (builtins.sort (a: b: getAttrLine a < getAttrLine b))
      ];
```